src/diffusers/models/transformer_2d.py

```diff
@@ -315,9 +338,20 @@ def forward(
             hidden_states = self.latent_image_embedding(hidden_states)
         elif self.is_input_patches:
             hidden_states = self.pos_embed(hidden_states)
+            if self.adaln_single is not None:
+                if added_cond_kwargs is None:
+                    raise ValueError("`added_cond_kwargs` cannot be None when using `adaln_single`.")
+                batch_size = hidden_states.shape[0]
+                timestep, embedded_timestep = self.adaln_single(
+                    timestep, added_cond_kwargs, batch_size=batch_size, hidden_dtype=hidden_states.dtype
```

Comment:
```suggestion
                    timestep, added_cond_kwargs, hidden_dtype=hidden_states.dtype
```
It's a bit weird to me that we have to pass the `batch_size`